admin_guide/node_problem_detector.adoc

```diff
@@ -0,0 +1,328 @@
+:toc: macro
+:toc-title:
+:prewrap!:
+
+toc::[]
+
+== Overview
+
+The node problem detector monitors the health of your nodes
+by finding certain problems and reporting these problems to the API server.
+The detector runs as a daemonset on each node.
+
+[IMPORTANT]
+====
+The node problem detector is a Technology Preview feature
+only.
+ifdef::openshift-enterprise[]
+Technology Preview features are not supported with Red Hat production service
+level agreements (SLAs), might not be functionally complete, and Red Hat does
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+The **node-problem-detector** daemon
+collects node problems from various daemons and make them visible to the upstream
+layers in the following categories:
+
+* `NodeCondition`: A permanent problem that makes the node unavailable for pods.
+* `Event`: A temporary problem that has limited impact on pod but is informative.
+
+.Sample Node Problem Detector Output with No Issues 
+----
+message: kernel has no deadlock
+reason: DockerHung
+status: false
+type: KernelDeadLock
+----
+
+.Sample Output for KernelDeadLock Condition
+----
+message: tesk docker:1234 blocked for more than 300 seconds
+reason: DockerHung
+status: true
+type: KernelDeadLock
+----
+
+Among the node problems that can be detected are:
+
+* Container runtime issues:
+** Unresponsive runtime daemon
+
+* Hardware issues:
+** Bad CPU 
+** Bad memory
+** Bad disk
+
+* Kernel issues
+** Kernel deadlock 
+** Corrupted file system
+** Unresponsive runtime daemon
+
+* Infrastructure daemon issues: 
+** NTP service outage
+
+If these problems are undetected, {product-title} 
+might scheduling pods to the bad nodes.
+
+{product-title} supports the following problem daemons. Installation of the daemone is beyond the scope of this document.  
+
+|===
+| Problem Daemon | NodeCondition | Description
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/kernel-monitor.json[KernelMonitor]
+| KernelDeadlock
+| A system log monitor monitors kernel log and reports problem according to predefined rules.
+
+| link:https://github.com/abrt/abrt/wiki[AbrtAdaptor]
+| None
+| Monitor ABRT log messages and report them further. ABRT (Automatic Bug Report Tool) is health monitoring daemon able 
+to catch kernel problems as well as application crashes of various kinds occurred on the host.
+
+| link:https://github.com/kubernetes/node-problem-detector/blob/master/config/custom-plugin-monitor.json[CustomPluginMonitor]
+| On-demand (based on user configuration)
+| A custom plugin monitor that checks various node problems with xref:admin-guide-node-problem-detector-monitor[custom, user-defined scripts].
+
```

Comment:
@joelsmith I have user-defined scripts linking to my "Using Custom Monitoring Tools" section below. Is this correct?